format/apple/bookmark/apple_bookmark.go

```diff
@@ -299,32 +300,12 @@ const (
 	dictEntrySize  = 4
 )
 
-type posLoopDetector []int64
```

Comment:
I was hoping to use generics for this but then discovered that Go doesn't support generics in method signatures ☹️  I don't know if there is a clever trick that I'm missing that could make this work well, but it would be nice if this could be generic for all type satisfying `constraints.Integer`.

Reply:
Yes it's a bit annoying limitation, but seems like a tricky thing to mix with interfaces :( is also the reason why i haven't used generics much for the decode code yet. Think i will stick with generating code instead for now, which also makes it very flexible... but becomes a bit complex.

In the case of `posLoopDetector` could the generics type be in on the type? something like:
```
type posLoopDetector[T constraints.Integer] []T

func (t *posLoopDetector[T]) a() {
	...
}
```
Or you mean something else?

Reply:
Thanks, that's pretty much what I was trying to do but I was putting the generic type in the wrong place.